lib/rspec/core.rb

```diff
@@ -178,4 +177,7 @@ def self.const_missing(name)
     require MODULES_TO_AUTOLOAD.fetch(name) { return super }
     ::RSpec.const_get(name)
   end
+
+  Core::DSL.expose_globally!
+  Core::SharedExampleGroup::TopLevelDSL.expose_globally!
```

Comment:
Whilst I get that this is to prevent configuration being loaded prematurely, feels a bit like duplication :/